### PR TITLE
Mosaic chart for nominal features

### PR DESCRIPTION
**Pull request by Speculative**
image

This adds a Mosaic chart if there are exactly two nominal features in the sample data. The sample `Breakdown` in `HighLevelStats` has been refactored to use this generic `MosaicChart` component.

### CODE REVIEW

File: webview-ui/src/panes/Charts.tsx [hunk 1]
```diff
@@ -11,46 +12,106 @@ type ChartsProps = {
...
+        setExampleFilter={setExampleFilter}
+        horizontalAxis={horizontalBuckets.map((bucket) => [
+          bucket,
+          (x) => x.features.nominal[horizontalFeature] === bucket,
+        ])}
+        verticalAxis={verticalBuckets.map((bucket) => [
+          bucket,
+          (x) => x.features.nominal[verticalFeature] === bucket,
+        ])}
+        colors={["accent", "accent2", "accent3", "warning", "primary"]}
```
Comment (Speculative) [inline webview-ui/src/panes/Charts.tsx:113]: I had trouble with colors. Ideally we'd have a bigger palette than 5. Currently if you have more than 5 rows, something bad will happen (no background color at all). We could add more colors, or I could make the colors repeat.

Comment (hgoldstein95) [reply webview-ui/src/panes/Charts.tsx:113]: I vote both, maybe? I'm using the [nord](https://www.nordtheme.com/docs/colors-and-palettes) color palette, so we can probably pull a few more colors from there, and then I think we should probably also have it repeat.

**Comment by Speculative** [comment]
Changes:

* Added extra colors, loop colors if # categories > # colors
* Dropdowns to select features if there are > 2
image
* If there are exactly 2 features, don't show the dropdowns
image

Currently I've made it so that you can't select the same feature in both dropdowns but this is a bit weird because the content of the dropdowns keeps changing. I can change it so that both dropdowns always have the same content, but then you can create weird charts of the same thing plotted against itself
image

**Comment by hgoldstein95** [comment]
I think it's fine to let you plot something against itself. No one will do it, and it keeps the UX a bit less frustrating for the common case. Excited to merge once that's done!

**Pull request merged**